### PR TITLE
Add support for font ligatures in Emacs 27 +

### PR DESCRIPTION
**Pull request by AdeAttwood**
Ligature support is provided by https://github.com/mickeynp/ligature.el

Ref: #6972

### CODE REVIEW

File: layers/+fonts/ligatures/packages.el [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    '((ligature :location (recipe :fetcher github :repo "mickeynp/ligature.el"))))
+  (defconst ligatures-packages '()))
+
+(defun ligatures/init-ligature ()
+  "Initialise the ligatures"
+  (dolist (mode ligature-modes)
+    (ligature-set-ligatures mode ligature-set))
+  (global-ligature-mode t))
+
+;;; packages.el ends here
```
Comment (smile13241324) [inline layers/+fonts/ligatures/packages.el:23]: No package statements please

**Review by smile13241324 — CHANGES_REQUESTED** [review]
Hi @AdeAttwood,

I have double checked your contribution. In principle I like it however I have two issues we need to solve before this is to be merged:
* In your code you activate `ligature-mode` globally, in your documentation however you mention that users can configure modes where it should be activated, can you double check that the configuration as described in the README really works? I would have expected that ligatures are on for all modes independent of the modes I configure.
* I don't think that this feature justifies its own layer. Rather I would like to have this integrated into the `unicode-fonts` layer. There it could be activated with a configuration variable.

**Comment by AdeAttwood** [comment]
@smile13241324 thanks for the feedback.

I have moved the ligature code out of its own layer and moved it under the unicode-fonts

I have checked and the `unicode-fonts-ligature-modes` and `unicode-fonts-enable-ligatures` work as described in the README. I have tested in Emacs 26.3 and 27.1

**Review by smile13241324 — APPROVED** [review]
Looks fine now, will be merged soon.

**Comment by smile13241324** [comment]
Thank you for contributing to spacemacs 💜, I have cherry picked your commit into develop.

**Pull request merged**